tools/integration_tests/grpc_validation/setup_test.go

```diff
@@ -0,0 +1,166 @@
+	detectedAttrs, err := resource.New(ctx, resource.WithDetectors(gcp.NewDetector()))
+	if err != nil {
+		log.Printf("Error fetching the test environment.All tests will be skipped. Error : %v", err)
+		return "", err
+	}
+	attrs := detectedAttrs.Set()
+	if v, exists := attrs.Value("gcp.gce.instance.hostname"); exists && strings.Contains(strings.ToLower(v.AsString()), cloudtopProd) {
+		return cloudtopProd, nil
+	}
+	if v, exists := attrs.Value("cloud.region"); exists {
+		return v.AsString(), nil
+	}
+	return "", nil
+}
+
+// For testing with single region buckets, we need to find the region for success case.
+// If the input is 'us-west1-a' which is the test VM Zone, then this function returns 'us-west1' used
+// creating the test buckets.
+func findSingleRegionForGRPCDirectPathSuccessCase(testRegion string) string {
+	parts := strings.Split(testRegion, "-")
+	if len(parts) >= 2 {
+		return strings.Join(parts[:len(parts)-1], "-") //rejoin the first parts of the zone, excluding the last part.
+	}
+	return ""
+}
+
+// For testing with multi region buckets, we need to find the region for success case.
+// If the input is 'us-west1-a' which is the test VM Zone, then this function returns 'us' used
+// creating the test buckets.
+func findMultiRegionForGRPCDirectPathSuccessCase(testRegion string) string {
+	parts := strings.Split(testRegion, "-")
+	if len(parts) > 0 {
+		return parts[0] // Return the first part of the string i.e. if us-central1 then us
+	}
+	return ""
+}
+
+// Generic function to pick a region other than the passed value to validate for failure scenario.
+func pickFailureRegionFromListOfRegions(successRegion string, regions []string) string {
+	for _, otherRegion := range regions {
+		if otherRegion != successRegion {
+			return otherRegion
+		}
+	}
+	return ""
+}
+
+// Creating test bucket name with unique suffix.
+func createTestBucketName(region string) string {
+	epoch := time.Now().UnixNano() // Get the current Unix epoch time
+	return fmt.Sprintf("grpc_validation_%s_%d", region, epoch)
+}
+
+// Based on the test case, we need to create the bucket in/out of region with the test VM.
+// This has to be done dynamically at the time of test setup.
+// Based on what region we pass, the test bucket will be multi region or single region.
+func createTestBucket(testBucketRegion, testBucketName string) (err error) {
+	bucket := client.Bucket(testBucketName)
+	if err = bucket.Create(ctx, gcp_project, &storage.BucketAttrs{Location: testBucketRegion}); err != nil {
+		log.Printf("Error while creating bucket, error: %v", err)
+		return err
+	}
+	return nil
+}
+
+func TestMain(m *testing.M) {
+	// Parse flags from the setup.
+	var err error
+	setup.ParseSetUpFlags()
+	if err := setup.SetUpTestDir(); err != nil {
+		log.Fatalf("Failed to setup GCSFuse package. Error: %v", err)
+	}
+
+	// Creating a common storage client for the test
+	ctx = context.Background()
+	if client, err = storage.NewClient(ctx); err != nil {
```

Comment:
Can we use [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/client/storage_client.go#L43) function it self